### ISSUE TITLE
In modules.neuron.py

### ISSUE THREAD

**Issue by ghost** [qymeng94/DSR#2]
Line 148:   Vth_grad[input_rate_coding <= Vth / delta_t] = 0
Should if be modified to: Vth_grad[input_rate_coding <= Vth / delta_t * tau] = 0 ? (According to the fomula 14 in the paper)

**Comment by qymeng94** [comment]
Thank you for pointing it out! I have already modified the code. BTW, since we set tau = 1 for all the experiments, the results are not affected with or without the modification.

**Comment by ghost** [comment]
Line 149: Vth_grad = torch.sum(Vth_grad) * delta_t
Should if be modified to: Vth_grad = torch.sum(Vth_grad) / delta_t ? (According to the fomula 14 in the paper)

**Comment by qymeng94** [comment]
Thank you very much for your attention. By formula 14, there should be a division, not a multiplication. However, the "correct" gradients would fail the training in our experiments: thresholds change too fast during training and tend to be 0 after several epochs. So we shrink the gradients by a factor of delta_t^2. 
We also explained this trick in the paper: "In practice, since we use mini-batch optimization methods to train SNNs, the gradient for each threshold is proportional to the batch size by the chain rule. Thus, we scale the gradient regarding different batch sizes and spiking neural models."
Why do we scale the gradients by delta_t^2? We find that the training for the IF neuron is fine without Vth_grad scaling. So we want Vth_grad/Vth for IF and LIF to have the same order, and using the scaling factor delta_t^2 for LIF can achieve this.

**Comment by ghost** [comment]
Thank you very much for your explanation.
